app/models/plan.rb

```diff
@@ -174,7 +174,8 @@ class Plan
   scope :by_plan_type,          ->(plan_type) { where(plan_type: plan_type) }
   scope :by_dental_level_for_bqt,       ->(dental_level) { where(:dental_level.in => dental_level) }
   scope :by_plan_type_for_bqt,          ->(plan_type) { where(:plan_type.in => plan_type) }
-  scope :for_service_areas,         ->(service_areas) { where(service_area_id: { "$in" => service_areas }) }
+  scope :for_service_areas_and_carriers,->(service_areas, carrier_profile_ids) { where(service_area_id: { "$in" => service_areas }, carrier_profile_id: { "$in" => carrier_profile_ids}) }
```

Comment:
Again, this will produce the 'any combination matches' rather than the 'must be one of these pairs' behaviour we meant.